app/grandchallenge/challenges/signals.py

```diff
@@ -0,0 +1,65 @@
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from django.utils.timezone import timedelta
+
+from grandchallenge.challenges.models import Challenge, OnboardingTask
+
+
+@receiver(post_save, sender=Challenge)
```

Comment:
Signals should only be used as a last resort and are not necessary here. This is to prevent business logic being located in multiple locations to make it easier for future developers. See https://lincolnloop.com/insights/django-anti-patterns-signals/

There are already post save methods for `Challenges` in `models.py` and they should be extended with a new class method here: https://github.com/comic/grand-challenge.org/blob/bcaa9ed57915b07733eea065e7d17284f2218630/app/grandchallenge/challenges/models.py#L505

Reply:
Thanks for the link to the blog. It was a good read and I agree with its argument.